doc/api/vm.md

```diff
@@ -1573,6 +1573,105 @@ inside a `vm.Context`, functions passed to them will be added to global queues,
 which are shared by all contexts. Therefore, callbacks passed to those functions
 are not controllable through the timeout either.
 
+## Local Worker
+
+> Stability: 1 - Experimental
+
+### Class: `LocalWorker`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+A `LocalWorker` is effectively a Node.js environment that runs within the
+same thread.
+
+```mjs
+import { LocalWorker } from 'vm';
+import { fileURLToPath } from 'url';
+const w = new LocalWorker();
+const myAsyncFunction = w.createRequire(fileURLToPath(import.meta.url))('my-module');
+console.log(await myAsyncFunction());
+```
+
+#### `new LocalWorker()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `localworker.runInWorkerScope(fn)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function}
+
+Wrap `fn` and run it as if it were run on the event loop of the inner Node.js
+instance. In particular, this ensures that Promises created by the function
+itself are resolved correctly. You should generally use this to run any code
+inside the inner Node.js instance that performs asynchronous activity and that
+is not already running in an asynchronous context (you can compare this to
+the code that runs synchronously from the main file of a Node.js application).
```

Comment:
So this allows me to run asynchronous code synchronously? Or is it only `createRequire` that returns a synchronous function like `require`, and `createImport` returns an async function like `import()`?

```js
import { LocalWorker } from ‘vm’

const source = `
  const { version } = (await fetch('https://nodejs.org/dist/index.json').then(res => res.json()))[0]
  console.log('Latest Node.js version', version)
`
const moduleUrl = `data:text/javascript,${encodeUriComponent(source)}`

const localWorker = new LocalWorker()
const getLatestNodeVersion = localWorker.createImport(import.meta.url)(moduleUrl)

// This?
const latestNodeVersion = getLatestNodeVersion()
// Or this?
const latestNodeVersion = await getLatestNodeVersion()
```

Reply:
> So this allows me to run asynchronous code synchronously

No, this is leftover docs that I would need to remove.